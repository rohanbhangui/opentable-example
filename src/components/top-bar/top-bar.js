import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { dataRequested } from '../../redux/actions';

import logo from '../../assets/images/logo.svg';
import './top-bar.scss';


const TopBar = ({ dataRequested }) => {
  const submitContent = loc => (e) => {
    let payload = {};

    if (e) {
      e.preventDefault();

      const formData = new FormData(e.target);
      const city = formData.get('location');
      const name = formData.get('refine');


      if (city) {
        payload = { ...payload, city };
      }

      if (name) {
        payload = { ...payload, name };
      }
    }

    if (loc) {
      payload = { ...payload, city: loc };
    }

    dataRequested(payload);
  };

  useEffect(() => {
    submitContent('toronto')();
  }, []);

  return (
    <div className="header-container">
      <form onSubmit={submitContent()}>
        <label htmlFor="venueType">
          <div>Refine</div>
          <input name="refine" type="text" placeholder="Restaraunts, Breakfast, etc." />
        </label>
        <span>in</span>
        <label htmlFor="location">
          <div>Location</div>
          <input name="location" required type="text" placeholder="Toronto, Boston, New York, etc." />
        </label>
        <input type="submit" value="Search" />
      </form>
    </div>
  );
};

export default connect(null, { dataRequested })(TopBar);
