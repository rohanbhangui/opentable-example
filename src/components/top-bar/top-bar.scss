@import '../../assets/styles/_variables.scss';

header {
	position: sticky;
	top: 0;
	background: $color-primary;
	border-bottom: 1px solid darken($color-primary, 20%);
	z-index: 400;

	.header-container {
		display: flex;
		padding: 0.3rem 0.8rem;
		width: 100%;
		max-width: 85.375rem;
		margin: 0 auto;
    align-items: center;
    justify-content: center;

		@include small {
			padding: 1rem 0.8rem;
		}

		form {
			color: white;

			> * {
				display: inline-block;
			}

			> span {
				margin: 0 .5rem;
			}

			input {
				font-size: 1rem;
				padding: .5rem;
				border-radius: .25rem;

				&[type="submit"] {
					background: black;
					color: white;
					margin-left: .5rem;
					border: 1px solid #000;
				}
			}
			
			label {
				div {
					font-size: .8rem;
					display: block;
					margin-bottom: .2rem;
					font-weight: bold;
				}

				input {
					&[type="text"] {	
						background: rgba(255, 255, 255, .33);
						transition: ease .2s;
						border: 1px solid rgba(0, 0, 0, 0);
						min-width: 12.5rem;
						color: #fff;

						&::placeholder {
							color: rgba(255, 255, 255, .33);
						}

						
						&:focus {
							background: white;
							border: 1px solid #999;
							color: #000;
							
							&::placeholder {
								color: #999;
							}
						}
					}
				}
			}
		}
	}
}