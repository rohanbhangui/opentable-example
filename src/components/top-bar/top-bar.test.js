import React from 'react';
import { shallow } from 'enzyme';
import { TopBar } from './top-bar';

describe('TopBar', () => {
  let component;
  const mockDataRequestedFn = jest.fn();

  beforeEach(() => {
    component = shallow(<TopBar dataRequested={mockDataRequestedFn} />);
  });

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the mock submission function', () => {
    component.find('#search').simulate(
      'submit',
      { preventDefault() {} },
    );
    expect(mockDataRequestedFn.mock.calls.length).toBe(1);
  });

  it('should be called with a name value and city value', () => {
    component.find('#city').simulate(
      'change',
      { target: { name: 'city', value: 'yyy' } },
    );

    component.find('#search').simulate(
      'submit',
      { preventDefault() {} },
    );

    expect(mockDataRequestedFn.mock.calls[1][0]).toEqual(
      { city: 'yyy', refine: '' },
    );

    component.find('#refine').simulate(
      'change',
      { target: { name: 'refine', value: 'xxx' } },
    );

    component.find('#search').simulate(
      'submit',
      { preventDefault() {} },
    );

    expect(mockDataRequestedFn.mock.calls[2][0]).toEqual(
      { refine: 'xxx', city: '' },
    );
  });
});
