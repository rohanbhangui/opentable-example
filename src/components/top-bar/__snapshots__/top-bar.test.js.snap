// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render 1`] = `ShallowWrapper {}`;
